### DISCUSSION TITLE
bootstrap-vcpkg should not rebuild vcpkg if nothing has changed

### DISCUSSION THREAD

**Discussion by Gerhard-Herbert** [microsoft/vcpkg#17831] [Ideas]
****

**bootstrap-vcpkg always rebuilds vcpg executable**
Port updates and source-code  of vcpkg are in the same repository. Our build-system is updating the repository regularly and sometimes vcpkg is not working anymore. It seems like vcpkg source-code has changed and a rebuild is needed. Therefore we added the bootstrap-vcpkg script to the build. This solved the out-of-date problem but now vcpkg is rebuild very often.

**Proposed solution**
Just rebuild vcpkg if the repository has changed.

**Describe alternatives you've considered**
Store the has of the repository and run bootstrap-vcpkg only if the hash has changed.

**Comment by JackBoosY** [comment]
Related: #13738.

**Comment by JackBoosY** [comment]
Since vcpkg published binary executable on Windows, this issue just related to non-Windows.
